### DISCUSSION TITLE
How to calculate strings or Floating-point arithmetic numbers？

### DISCUSSION THREAD

**Discussion by yongxin3344520** [Z3Prover/z3#6837] [Q&A]
How to calculate strings or Floating-point arithmetic numbers like this?

Example 1：
     // String a ;
    a.length() == 2 && a.startWith("@")

Example 2：
     // float x, y ;
    x + y =  1; 
    x^2 + y^2 = 2 ;

**Comment by NikolajBjorner** [comment]
Try to follow https://microsoft.github.io/z3guide/programming/Z3%20Python%20-%20Readonly/Introduction
Bing Chat or OpenAI and probably other services provide fairly useful answers to such questions too.
I tried your question on Bing Chat and got very reasonable leads.
